api/app/controllers/api/v1/ingests_controller.rb

```diff
@@ -0,0 +1,53 @@
+module API
+  module V1
+    # Trigger ingestion processing via the API
+    class IngestsController < ApplicationController
+      before_action :set_project, only: [:create]
+      before_action :set_text, only: [:create]
+
+      resourceful! Ingestion do
+        @project&.ingestions
+        @text&.ingestions
+      end
+
+      # rubocop:disable Metrics/AbcSize, Metrics/MethodLength
+      def create
```

Comment:
To answer your larger question, I would make this a separate controller. By all means, you could make a concern to DRY things up, but the controller should only handle one kind of ingestion parent. You had basically something perfect, I just broke it out into a gist because typing it up in these little windows isn't a good time:

https://gist.github.com/scryptmouse/75d4d1cbd8f487bfa68a2594631e2177

Reply:
The biggest reason is the SRP (single responsibility principle). Even though the concern is almost the same code, the difference is you're specifying that for `api/v1/projects/ingestions_controller.rb` it's handling project ingestions, and for `api/v1/texts/ingestions_controller.rb` it's handling text ingestions.